tests/integrations/config/test_config_client.rs

```diff
@@ -0,0 +1,256 @@
+    where
+        F: Fn(&mut String),
+    {
+        let mut configs = self.configs.lock().unwrap();
+        let cfg = configs.get_mut(id).unwrap();
+        f(&mut cfg.content);
+        cfg.version.local += 1;
+    }
+
+    fn get(&self, id: &str) -> Config {
+        self.configs.lock().unwrap().get(id).unwrap().clone()
+    }
+}
+
+impl PdClient for MockPdClient {
+    fn register_config(&self, id: String, v: Version, cfg: String) -> Result<CreateResponse> {
+        let old = self
+            .configs
+            .lock()
+            .unwrap()
+            .insert(id.clone(), Config::new(v.clone(), cfg.clone(), Vec::new()));
+        assert!(old.is_none(), format!("id {} already be registered", id));
+
+        let mut status = Status::default();
+        status.set_code(StatusCode::Ok);
+        let mut resp = CreateResponse::default();
+        resp.set_status(status);
+        resp.set_config(cfg);
+        resp.set_version(v);
+        Ok(resp)
+    }
+
+    fn get_config(&self, id: String, version: Version) -> Result<GetResponse> {
+        let mut resp = GetResponse::default();
+        let mut status = Status::default();
+        let configs = self.configs.lock().unwrap();
+        if let Some(cfg) = configs.get(&id) {
+            match cmp_version(&cfg.version, &version) {
+                Ordering::Equal => status.set_code(StatusCode::NotChange),
+                _ => {
+                    resp.set_config(cfg.content.clone());
+                    status.set_code(StatusCode::WrongVersion);
+                }
+            }
+            resp.set_version(cfg.version.clone());
+        } else {
+            status.set_code(StatusCode::Unknown);
+        }
+        resp.set_status(status);
+        Ok(resp)
+    }
+
+    fn update_config(
+        &self,
+        id: String,
+        version: Version,
+        mut entries: Vec<ConfigEntry>,
+    ) -> Result<UpdateResponse> {
+        let mut resp = UpdateResponse::default();
+        let mut status = Status::default();
+        if let Some(cfg) = self.configs.lock().unwrap().get_mut(&id) {
+            match cmp_version(&cfg.version, &version) {
+                Ordering::Less => {
+                    cfg.update.append(&mut entries);
+                    cfg.version = version;
+                    status.set_code(StatusCode::Ok);
+                }
+                _ => status.set_code(StatusCode::WrongVersion),
+            }
+            resp.set_version(cfg.version.clone());
+        } else {
+            status.set_code(StatusCode::Unknown);
+        }
+        resp.set_status(status);
+        Ok(resp)
+    }
+}
+
+fn validated_cfg() -> TiKvConfig {
+    let mut cfg = TiKvConfig::default();
+    cfg.validate().unwrap();
+    cfg
+}
+
+#[test]
+fn test_update_config() {
+    let pd_client = Arc::new(MockPdClient::new());
+    let id = "localhost:1080";
+
+    // register config
+    let mut cfg_handler = pd_client.clone().register(id, validated_cfg());
```

Comment:
What if it registers twice?

Reply:
Which depend on how PD implement `Create`, we can't test that behavior here since `pd_client` here are implemented by ourself.